EIPS/eip-3643.md

```diff
@@ -181,21 +163,28 @@ interface IERC3643 is IERC20 {
     function batchSetAddressFrozen(address[] calldata _userAddresses, bool[] calldata _freeze) external;
     function batchFreezePartialTokens(address[] calldata _userAddresses, uint256[] calldata _amounts) external;
     function batchUnfreezePartialTokens(address[] calldata _userAddresses, uint256[] calldata _amounts) external;
-
-    // roles setting
-    function transferOwnershipOnTokenContract(address _newOwner) external;
-    function addAgentOnTokenContract(address _agent) external;
-    function removeAgentOnTokenContract(address _agent) external;
 }
 
 ```
 
 ### Identity Registry Interface
 
-This Identity Registry is linked to storage that contains a dynamic whitelist of identities. The Identity Registry makes the link between a wallet address, an [ONCHAINID](https://tokeny.com/onchainid/) and a country code corresponding to the country of residence of the investor, this country code is set in accordance with the [ISO-3166 standard](https://www.iso.org/iso-3166-country-codes.html). It also contains a function called `isVerified()`, which returns a status based on the validity of claims (as per the security token requirements) in the user’s ONCHAINID. The Identity Registry is managed by the agent wallet(s) i.e. only the agent(s) can add or remove identities in the registry (note: the agent role on the Identity Registry is set by the owner, therefore the owner could set himself as the agent if he wants to keep everything under his own control). There is a specific identity registry for each security token. 
-A detailed description of the functions can be found in the [interfaces folder](https://github.com/TokenySolutions/EIP3643/tree/main/interfaces). 
-
-Note that [`IClaimIssuer`](https://github.com/onchain-id/solidity/blob/master/contracts/interface/IClaimIssuer.sol) and [`IIdentity`](https://github.com/onchain-id/solidity/blob/master/contracts/interface/IIdentity.sol) are needed in this interface and are coming from [ONCHAINID](https://github.com/onchain-id/solidity)
+This Identity Registry is linked to storage that contains a dynamic whitelist of identities. The Identity Registry 
+makes the link between a wallet address, an Identity smart contract and a country code corresponding to the country of 
+residence of the investor, this country code is set in accordance with the ISO-3166 standard. It also contains a 
+function called `isVerified()`, which returns a status based on the validity of claims (as per the security token 
+requirements) in the user’s Identity contract.
+In the T-REX implementation of this standard, we decided to use the Ownable library from OpenZeppelin as well as an 
+Agent Role implemented with the [Roles Library](../assets/eip-3643/roles/Roles.sol) to manage access to functions on 
```

Comment:
EIPs should only specify the externally visible properties of a smart contract. They should not specify _how_ an implementation implements those properties.

Reply:
done in https://github.com/ethereum/EIPs/pull/7164/commits/97819ae191f6b2a9c9feaeb92c8d35214853b2de 
I removed all mentions of implementation, preferring the reference to ERC-173 for ownership, i added the interface `IAgentRole` to the specifications as the agent role is very crucial for the standard. A more granular approach of the Agent Role can be done through an external contract for better role management but it seems important to define the 2 main roles, `owner` and `agent`